### PR TITLE
Expose CanardRxSubscription

### PR DESCRIPTION
**Pull request by PetervdPerk-NXP**
As discussed in https://forum.uavcan.org/t/libcanard-listing-active-canardrxsubscription/1266

### CODE REVIEW

File: libcanard/canard.h [hunk 1]
```diff
@@ -268,7 +268,7 @@ typedef struct
 /// This is an intentional time-memory trade-off: use a large look-up table to ensure predictable temporal properties.
 typedef struct CanardRxSubscription
 {
-    struct CanardRxSubscription* _next;  ///< Internal use only.
+    struct CanardRxSubscription* next;  /// Read-only DO NOT MODIFY THIS
```
Comment (pavel-kirienko) [inline libcanard/canard.h:271]:
```suggestion
    struct CanardRxSubscription* next;  ///< Read-only DO NOT MODIFY THIS
```

This angle bracket is a Doxygen thing, let's keep it.

**Review by pavel-kirienko — CHANGES_REQUESTED** [review]
Should we not also change these?

https://github.com/UAVCAN/libcanard/blob/e7fcf5145dfff791dc5e7bb730b2f01a78ca6c9d/libcanard/canard.h#L288-L290

**Comment by pavel-kirienko** [comment]
Please run `make format` in your CMake binary dir to fix CI

**Comment by PetervdPerk-NXP** [comment]
> Please run `make format` in your CMake binary dir to fix CI

Try that, which version of clang-format is the CI runing maybe that's the issue.

```
hovergames@hovergames:/tmp/libcanard/tests/build$ make format
Formatting /tmp/libcanard/tests/../libcanard/canard.c
YAML:15:38: error: invalid boolean
AllowShortIfStatementsOnASingleLine: Never
                                     ^~~~~
Error reading /tmp/libcanard/tests/../.clang-format: Invalid argument
Formatting /tmp/libcanard/tests/../libcanard/canard.h
YAML:15:38: error: invalid boolean
AllowShortIfStatementsOnASingleLine: Never
                                     ^~~~~
Error reading /tmp/libcanard/tests/../.clang-format: Invalid argument
Formatting /tmp/libcanard/tests/../libcanard/canard_dsdl.c
YAML:15:38: error: invalid boolean
AllowShortIfStatementsOnASingleLine: Never
                                     ^~~~~
Error reading /tmp/libcanard/tests/../.clang-format: Invalid argument
Formatting /tmp/libcanard/tests/../libcanard/canard_dsdl.h
YAML:15:38: error: invalid boolean
AllowShortIfStatementsOnASingleLine: Never
                                     ^~~~~
Error reading /tmp/libcanard/tests/../.clang-format: Invalid argument
Formatting /tmp/libcanard/tests/exposed.hpp
YAML:15:38: error: invalid boolean
AllowShortIfStatementsOnASingleLine: Never
                                     ^~~~~
Error reading /tmp/libcanard/.clang-format: Invalid argument
Formatting /tmp/libcanard/tests/helpers.hpp
YAML:15:38: error: invalid boolean
AllowShortIfStatementsOnASingleLine: Never
                                     ^~~~~
Error reading /tmp/libcanard/.clang-format: Invalid argument
Formatting /tmp/libcanard/tests/test_dsdl.cpp
YAML:15:38: error: invalid boolean
AllowShortIfStatementsOnASingleLine: Never
                                     ^~~~~
Error reading /tmp/libcanard/.clang-format: Invalid argument
Formatting /tmp/libcanard/tests/test_private_crc.cpp
YAML:15:38: error: invalid boolean
AllowShortIfStatementsOnASingleLine: Never
                                     ^~~~~
Error reading /tmp/libcanard/.clang-format: Invalid argument
Formatting /tmp/libcanard/tests/test_private_rx.cpp
YAML:15:38: error: invalid boolean
AllowShortIfStatementsOnASingleLine: Never
                                     ^~~~~
Error reading /tmp/libcanard/.clang-format: Invalid argument
Formatting /tmp/libcanard/tests/test_private_tx.cpp
YAML:15:38: error: invalid boolean
AllowShortIfStatementsOnASingleLine: Never
                                     ^~~~~
Error reading /tmp/libcanard/.clang-format: Invalid argument
Formatting /tmp/libcanard/tests/test_public_roundtrip.cpp
YAML:15:38: error: invalid boolean
AllowShortIfStatementsOnASingleLine: Never
                                     ^~~~~
Error reading /tmp/libcanard/.clang-format: Invalid argument
Formatting /tmp/libcanard/tests/test_public_rx.cpp
YAML:15:38: error: invalid boolean
AllowShortIfStatementsOnASingleLine: Never
                                     ^~~~~
Error reading /tmp/libcanard/.clang-format: Invalid argument
Formatting /tmp/libcanard/tests/test_public_tx.cpp
YAML:15:38: error: invalid boolean
AllowShortIfStatementsOnASingleLine: Never
                                     ^~~~~
Error reading /tmp/libcanard/.clang-format: Invalid argument
Formatting /tmp/libcanard/tests/test_self.cpp
YAML:15:38: error: invalid boolean
AllowShortIfStatementsOnASingleLine: Never
                                     ^~~~~
Error reading /tmp/libcanard/.clang-format: Invalid argument
CMakeFiles/format.dir/build.make:74: recipe for target 'CMakeFiles/format' failed
make[3]: *** [CMakeFiles/format] Error 1
CMakeFiles/Makefile2:277: recipe for target 'CMakeFiles/format.dir/all' failed
make[2]: *** [CMakeFiles/format.dir/all] Error 2
CMakeFiles/Makefile2:284: recipe for target 'CMakeFiles/format.dir/rule' failed
make[1]: *** [CMakeFiles/format.dir/rule] Error 2
Makefile:224: recipe for target 'format' failed
make: *** [format] Error 2
hovergames@hovergames:/tmp/libcanard/tests/build$ 

```

**Comment by pavel-kirienko** [comment]
The docs say that you need v11 or newer, yes: https://github.com/UAVCAN/libcanard/blob/master/CONTRIBUTING.md#tools

**Comment by PetervdPerk-NXP** [comment]
Pushed a new commit, although clang-format-12 seems to be changing formatting of other parts as well

https://github.com/UAVCAN/libcanard/pull/167/files#diff-2cf2cbe80f83756b41a08c76c014cc84c5ea7a2416150e5e45c0c1c4b4986da6R70

**Comment by pavel-kirienko** [comment]
Sigh. Can you try v11 then, please? A better option though is to update the settings to prevent breakage due to different versions having different default settings.

**Comment by PetervdPerk-NXP** [comment]
Using clang-format-11 solves it indeed.

**Review by pavel-kirienko — APPROVED** [review]

**Pull request merged**